### PR TITLE
docs: show creation of DFSchema

### LINKED ISSUE #8131: Add quick example of creating unqualified DFSchema
### Is your feature request related to a problem or challenge?

As a newer user, it wasn't obvious to me how to create an unqualified DFSchema from an Arrow schema.

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by wjones127**
## Which issue does this PR close?

Closes #8131.

## Rationale for this change

I got stuck figuring this out, and noticed there was very little documentation here.

I don't think I yet understand enough about DFSchema to fully document it. For now I'm mostly using unqualified schemas. Therefore, I'm not trying to complete #7724 with this PR.

## What changes are included in this PR?

Adds some initial examples.

## Are these changes tested?

Examples run with doc tests.

### CODE REVIEW

**Comment by alamb** [comment]
Thank you very much @wjones127

File: datafusion/common/src/dfschema.rs [hunk 1]
```diff
@@ -34,10 +34,75 @@ use crate::{
...
 pub type DFSchemaRef = Arc<DFSchema>;
 
-/// DFSchema wraps an Arrow schema and adds relation names
+/// DFSchema wraps an Arrow schema and adds relation names.
+///
+/// The schema may hold the fields across multiple tables. Some fields may be
+/// qualified and some unqualified. A qualified field is a field that has a
+/// relation name associated with it.
+///
+/// Unqualified fields must unique not only amongst themselves, but also must
```
Comment (andygrove) [inline datafusion/common/src/dfschema.rs:46]:
```suggestion
/// Unqualified fields must be unique not only amongst themselves, but also must
```

File: datafusion/common/src/dfschema.rs [hunk 2]
```diff
@@ -34,10 +34,75 @@ use crate::{
...
+/// have a distinct name from any qualified field names. This allows finding a
+/// qualified field by name to be possible, so long as there aren't multiple
+/// qualified fields with the same name.
+///
+/// There is an alias to `Arc<DFSchema` named [DFSchemaRef].
```
Comment (andygrove) [inline datafusion/common/src/dfschema.rs:51]:
```suggestion
/// There is an alias to `Arc<DFSchema>` named [DFSchemaRef].
```

**Review by andygrove — APPROVED** [review]
This is very helpful. Thanks @wjones127

File: datafusion/common/src/dfschema.rs [hunk 3]
```diff
@@ -34,10 +34,75 @@ use crate::{
...
+/// let arrow_schema = Schema::new(vec![
+///    Field::new("c1", DataType::Int32, false),
+/// ]);
+///
+/// let df_schema = DFSchema::try_from(arrow_schema).unwrap();
+/// let column = Column::new_unqualified("c1");
+/// assert!(df_schema.has_column(&column));
+/// ```
+///
+/// # Converting back to Arrow schema
```
Comment (alamb) [inline datafusion/common/src/dfschema.rs:92]: Thank you @wjones127  this is super helpful

**Review by alamb — APPROVED** [review]
Thank you for this contribution @wjones127  -- I think this will help avoid a bunch of confusion

**Pull request merged**